solr/core/src/test/org/apache/solr/response/transform/TestChildDocTransformerHierarchy.java

```diff
@@ -297,6 +297,35 @@ public void testNoChildren() throws Exception {
         "/response/docs/[0]/type_s==cake");
   }
 
+  @Test
+  public void testChildReturnFields() throws Exception {
+    indexSampleData(numberOfDocsPerNestedTest);
+
+    assertJQ(req("q", "test_s:testing",
+        "sort", "id asc",
+        "fl", "*, [child childFilter='lonely/lonelyGrandChild/test2_s:secondTest' fl='*, _nest_path_']",
```

Comment:
Lets leave `_nest_path_` as internal and not mention it since testing "fl" has nothing to do with this field or nested documents.  So instead use some other field, foo_s or whatever.  And can you move this test to TestChildDocTransformer.java as it's not related to hierarchy?

Reply:
Sure thing,
just pushed a new commit :).